### DISCUSSION TITLE
SCANN: Testing the Relationship between Search Parameters and Topk for Achieving a Balance between Performance and Recall

### DISCUSSION THREAD

**Discussion by Yang1032** [milvus-io/milvus#30854] [Q&A and General discussion]
Hi everyone,
I'm currently experimenting with the SCANN algorithm. I would like to know if any of you have tested the relationship between the search parameters, specifically nprobe and rerorder_k, in conjunction with the topk parameter. My goal is to find a balance between performance and recall.
Thank you in advance!

**Comment by yhmo** [comment]
SCANN is similar to IVF_PQ, and SCANN is more efficient.
Increase nprobe, the recall rate increases, and latency also increases.
reorder_k must be greater than topk, it uses fast scan algorithm to find out reorder_k items, then comptes real distance of the reorder_k vectors to find out the final topk items. 
With reorder_k increases, the recall rate increases, and latency also increases. Typically set reorder_k =10 * topk is ok, no need to set a large value because the recall rate increases very slowly when the reorder_k becomes greater.

Data also affects recall rate, different dataset gets different recall rate with the same nprobe/reorder_k. We don't have a theory to "find a balance between performance and recall".

**Reply by Yang1032** [reply]
Thanks, I'll give it another shot!
